chapters/annotations.tex

```diff
@@ -43,14 +43,19 @@ \section{Annotations for Documentation}\doublelabel{annotations-for-documentatio
 
 \begin{lstlisting}[language=grammar]
 documentation-annotation:
-   annotation "(" Documentation "(" "info" "=" STRING ["," "revisions" "=" STRING ] ")" ")"
+   annotation "(" Documentation "(" [ "info" "=" STRING ]
```

Comment:
This does not work, since `,` is a separator and if `info `is a missing there should be no `,` before `revisions`, and if both `info` and `revisions` are missing we should not have `, figures=…`. The more complicated question is: what works?
One variant would be:
```
documentation-annotation: annotation "(" Documentation "(" doc-item {", " doc-item} ")" ")"
doc-item: "info" "=" STRING |
                 "revisions" "=" STRING |
                 "figures" "=" figure-annotations
```
Another would be to use a record to describe it instead.

Reply:
The change only follows the existing style used for `revisions`, so unless `revisions` was only meant to be allowed when there is an `info` this should be fixed in a separate PR.  If `revisions` was only meant to be present when there is an `info`, your proposal should be changed accordingly (although I'm afraid one would need some extra lookahead in the parser to handle it):
```
doc-item: "info" "=" STRING [, "revisions" "=" STRING] |
                 "figures" "=" figure-annotations
```

Besides, I'd say that the `revisions` is very poorly documented:
> the `revisions` annotation giving a list of revisions and other annotations defined by a tool [_The `revisions` documentation may be omitted in printed documentation_].

It doesn't even seem to match current practice in the MSL, where it is set to a string, not a list.

Reply:
@HansOlsson, should we go for the pseudo code record to resolve this?

Reply:
That's likely the best solution if we want something formal.
Regarding the previous issues: I believe that revisions was almost always combined with info so in practice that made sense, and I think "list" mean that the html-code contain a list of revisions - not that revision-annotation itself was a list.

Reply:
To be resolved by #2594.